modules/salary/view/record.inc

```diff
@@ -176,13 +176,9 @@ elseif($_lib['input']->getProperty('action_defconf_save'))
 }
 elseif($_lib['input']->getProperty('action_salary_new'))
 {
-    list($JournalID, $tmp) = 
-        $accounting->get_next_available_journalid(
-            array('available'=>true, 'update'=>true, 'type' =>'L'));
-
-    $query="select SalaryID from salary where JournalID=$JournalID";
-    if($debug) print "$query<br>\n";
+    $query="select MAX(JournalID) as MaxJournalID from salary";
```

Comment:
Shouldn't we select $row_salary->SalaryID id too?
We are using that on line 182.